src/detect-base64-decode.c

```diff
@@ -89,17 +89,21 @@ int DetectBase64DecodeDoMatch(DetectEngineThreadCtx *det_ctx, const Signature *s
 
     decode_len = MIN(payload_len, data->bytes);
 
+    DEBUG_VALIDATE_BUG_ON(decode_len < 0);
 #if 0
     printf("Decoding:\n");
     PrintRawDataFp(stdout, payload, decode_len);
 #endif
 
-    uint32_t consumed = 0, num_decoded = 0;
-    (void)DecodeBase64(det_ctx->base64_decoded, det_ctx->base64_decoded_len_max, payload,
-            decode_len, &consumed, &num_decoded, Base64ModeRFC4648);
-    det_ctx->base64_decoded_len = num_decoded;
-    SCLogDebug("Decoded %d bytes from base64 data.",
-        det_ctx->base64_decoded_len);
+    Base64Decoded *b64d =
+            Base64Decode(payload, decode_len, det_ctx->base64_decoded_len_max, Base64ModeRFC4648);
+    if (b64d != NULL) {
+        DEBUG_VALIDATE_BUG_ON(b64d->decoded_len >= (uint32_t)decode_len);
```

Comment:
@catenacyber I'm unsure about this cast. Could you please check?

Reply:
I realized I should be more worried about the unseen cast happening in the fn call above.. I have a better idea.